src/estimation/src/AttitudeMahonyFilter.cpp

```diff
@@ -0,0 +1,231 @@
+    stateBuffer(2) = m_initial_state.m_orientation(2);
+    stateBuffer(3) = m_initial_state.m_orientation(3);
+    stateBuffer(4) = m_initial_state.m_angular_velocity(0);
+    stateBuffer(5) = m_initial_state.m_angular_velocity(1);
+    stateBuffer(6) = m_initial_state.m_angular_velocity(2);
+    stateBuffer(7) = m_initial_state.m_gyroscope_bias(0);
+    stateBuffer(8) = m_initial_state.m_gyroscope_bias(1);
+    stateBuffer(9) = m_initial_state.m_gyroscope_bias(2);
+    return true;
+}
+
+bool iDynTree::AttitudeMahonyFilter::getInternalState(iDynTree::Span< double >& stateBuffer) const
+{
+    stateBuffer(0) = m_state.m_orientation(0);
+    stateBuffer(1) = m_state.m_orientation(1);
+    stateBuffer(2) = m_state.m_orientation(2);
+    stateBuffer(3) = m_state.m_orientation(3);
+    stateBuffer(4) = m_state.m_angular_velocity(0);
+    stateBuffer(5) = m_state.m_angular_velocity(1);
+    stateBuffer(6) = m_state.m_angular_velocity(2);
+    stateBuffer(7) = m_state.m_gyroscope_bias(0);
+    stateBuffer(8) = m_state.m_gyroscope_bias(1);
+    stateBuffer(9) = m_state.m_gyroscope_bias(2);
+    return true;
+}
+
+size_t iDynTree::AttitudeMahonyFilter::getInternalStateSize() const
+{
+    size_t size = 0;
+    size += m_state.m_orientation.size();
+    size += m_state.m_angular_velocity.size();
+    size += m_state.m_gyroscope_bias.size();
+    return size;
+}
+
+bool iDynTree::AttitudeMahonyFilter::getOrientationEstimateAsQuaternion(iDynTree::Quaternion& q)
+{
+    q = m_state.m_orientation;
+    return true;
+}
+
+bool iDynTree::AttitudeMahonyFilter::getOrientationEstimateAsRotationMatrix(iDynTree::Rotation& rot)
+{
+    rot = m_orientationInSO3;
+    return true;
+}
+
+bool iDynTree::AttitudeMahonyFilter::getOrientationEstimateAsRPY(iDynTree::RPY& rpy)
+{
+    rpy = m_orientationInRPY;
+    return true;
+}
+
+void iDynTree::AttitudeMahonyFilter::getParameters(iDynTree::AttitudeMahonyFilterParameters& params)
+{
+    params = m_params;
+}
+
+bool iDynTree::AttitudeMahonyFilter::setInternalState(iDynTree::Span< double >& stateBuffer)
+{
+    if ((size_t)stateBuffer.size() != getInternalStateSize())
+    {
+        iDynTree::reportError("AttitudeMahonyFilter", "setInternalState", "state size mismatch, using default state");
+        return false;
+    }
+    m_state.m_orientation(0) = stateBuffer(0);
+    m_state.m_orientation(1) = stateBuffer(1);
+    m_state.m_orientation(2) = stateBuffer(2);
+    m_state.m_orientation(3) = stateBuffer(3);
+    m_state.m_angular_velocity(0) = stateBuffer(4);
+    m_state.m_angular_velocity(1) = stateBuffer(5);
+    m_state.m_angular_velocity(2) = stateBuffer(6);
+    m_state.m_gyroscope_bias(0) = stateBuffer(7);
+    m_state.m_gyroscope_bias(1) = stateBuffer(8);
+    m_state.m_gyroscope_bias(2) = stateBuffer(9);
+    return true;
+}
+
+bool iDynTree::AttitudeMahonyFilter::setParameters(const iDynTree::AttitudeMahonyFilterParameters& params)
+{   m_params = params;
+    return true;
```

Comment:
Don't you need to call `useMagnetoMeterMeasurements` as done in the EKF variant?

Reply:
This is not required here, because there is only one variable that is involved in this case and this does not require an entire resizing and re-initialization of the buffers, like in the EKF variant. This [comment](https://github.com/robotology/idyntree/pull/516#discussion_r) has more details.